### DISCUSSION TITLE
Question with respect to `get_acquisition_function`

### DISCUSSION THREAD

**Discussion by jduerholt** [meta-pytorch/botorch#2089] [Q&A] [answered]
## Issue description

I am changing our strategies to use also in the MO case the method `get_acquisition_function` from botorch's `acquisition/factory.py`. In the case of `qEHVI` and `qLogEHVI` I have the feeling that the filtering of the `Y` values with respect to the reference point is missing. They are only filtered with respect to their feasibility under constraints but not with respect to the reference point before entering the partioning (https://github.com/pytorch/botorch/blob/64d0c6785f739541e8190ad28927f52ba63df021/botorch/acquisition/factory.py#L195). 

From my my point of view, something like this (non working pseudocode) is missing:

``` python
y = (y > ref_point).all(axis=-1)
```

Of course, the user could also filter `Y` this upfront before calling `get_acquisition_function`, but this appears very inconvenient for me, especially as the feasibility filtering is done within the method. Or is this not necessary at all, because the filtering happens in the partioning?

**Comment by dme65** [comment, marked as answer]
Hi @jduerholt,

> Or is this not necessary at all, because the filtering happens in the partitioning?

Correct, we don't have to filter based on the reference point as observations that don't dominate the reference point won't contribute to the hypervolume anyway (so filtering them out here will have no effect). This is different from the `constraints` since these correspond to additional outcome constraints and we need to make sure we don't include any infeasible observations in the hypervolume computations.

**Reply by jduerholt** [reply]
Thank you very much for the explanation!
